tests/filecheck/frontend/dialects/scf.py

```diff
@@ -0,0 +1,82 @@
+# RUN: python %s | filecheck %s
+
+from xdsl.frontend.exception import CodeGenerationException
+from xdsl.frontend.program import FrontendProgram
+from xdsl.frontend.context import CodeContext
+from xdsl.frontend.dialects.builtin import i1, i32, f32
+
+p = FrontendProgram()
+with CodeContext(p):
+    # CHECK:      %{{.*}} : !i32 = scf.if(%{{.*}} : !i1) {
+    # CHECK-NEXT:   %{{.*}} : !i32 = symref.fetch() ["symbol" = @x]
+    # CHECK-NEXT:   scf.yield(%{{.*}} : !i32)
+    # CHECK-NEXT: } {
+    # CHECK-NEXT:   %{{.*}} : !i32 = symref.fetch() ["symbol" = @y]
+    # CHECK-NEXT:   scf.yield(%{{.*}} : !i32)
+    # CHECK-NEXT: }
+    def test_if_expr(cond: i1, x: i32, y: i32) -> i32:
+        return x if cond else y
+
+    # CHECK:      scf.if(%{{.*}} : !i1) {
+    # CHECK-NEXT:   scf.yield()
+    # CHECK-NEXT: } {
+    # CHECK-NEXT:   scf.yield()
+    # CHECK-NEXT: }
+    def test_if_I(cond: i1):
+        if cond:
+            pass
+        else:
+            pass
+
+    # CHECK:       %{{.*}} : !i1 = symref.fetch() ["symbol" = @a]
+    # CHECK-NEXT:  scf.if(%{{.*}} : !i1) {
+    # CHECK-NEXT:   scf.yield()
+    # CHECK-NEXT: } {
+    # CHECK-NEXT:   %{{.*}} : !i1 = symref.fetch() ["symbol" = @b]
+    # CHECK-NEXT:   scf.if(%{{.*}} : !i1) {
+    # CHECK-NEXT:     scf.yield()
+    # CHECK-NEXT:   } {
+    # CHECK-NEXT:     %{{.*}} : !i1 = symref.fetch() ["symbol" = @c]
+    # CHECK-NEXT:     scf.if(%{{.*}} : !i1) {
+    # CHECK-NEXT:       scf.yield()
+    # CHECK-NEXT:     } {
+    # CHECK-NEXT:       scf.yield()
+    # CHECK-NEXT:     }
+    # CHECK-NEXT:     scf.yield()
+    # CHECK-NEXT:   }
+    # CHECK-NEXT:   scf.yield()
+    # CHECK-NEXT: }
+    def test_if_II(a: i1, b: i1, c: i1):
+        if a:
+            pass
+        elif b:
+            pass
+        elif c:
+            pass
+
+    # CHECK:      %{{.*}} : !i1 = symref.fetch() ["symbol" = @cond]
+    # CHECK-NEXT: scf.if(%{{.*}} : !i1) {
+    # CHECK-NEXT:   scf.yield()
+    # CHECK-NEXT: } {
+    # CHECK-NEXT:   scf.yield()
+    # CHECK-NEXT: }
```

Comment:
Not sure you followed the discussion in #727 but it seems that the else regions in MLIR can be empty, if they are optional. I am fine with the conversion you are proposing, just wanted to shoutout the MLIR supports a "more native" version of a single-handed if then what you seem to encode here.

Reply:
I guess the python AST works with these empty regions, so maybe it's not worth the effort of changing/analyzing the structure